File: fitrbackend/src/main/java/com/example/fitrbackend/model/WorkoutExercise.java
package com.example.fitrbackend.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;
/**
 * Represents an exercise performed in a workout session.
 */
@Getter
@Entity
public class WorkoutExercise {

    /**
     * Unique identifier for the exercise in the workout session.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    /**
     * The workout session that the exercise was performed in.
     */
    @ManyToOne
    @JoinColumn(name = "workout_session_id")
    @Setter
    private WorkoutSession workoutSession;

    /**
     * The exercise that was performed.
     */
    @ManyToOne
    @JoinColumn(name = "exercise_id")
    @Setter
    private Exercise exercise;

    /**
     * The measurement type of the exercise.
     */
    @Setter
    private MeasurementType measurementType;

    public WorkoutExercise() {}

    /**
     * Constructor for WorkoutExercise.
     *
     * @param workoutSession the workout session that the exercise was performed in
     * @param exercise the exercise that was performed
     * @param measurementType the measurement type of the exercise
     */
    public WorkoutExercise(WorkoutSession workoutSession, Exercise exercise, MeasurementType measurementType) {
        this.workoutSession = workoutSession;
        this.exercise = exercise;
        this.measurementType = measurementType;
    }
}

File: fitrfrontend/fitrfrontend/Models/WorkoutExercise.swift
//
//  WorkoutExercise.swift
//  fitrfrontend
//
//  Created by Ambrose Blay on 2/17/26.
//

import Foundation

/// Represents an exercise performed in a workout session.
struct WorkoutExercise: Codable, Identifiable {
    /// Unique identifier for the exercise in the workout session
    let id: Int64
    
    /// The workout session ID that the exercise was performed in
    var workoutSessionId: Int64
    
    /// The exercise ID that was performed
    var exerciseId: Int64
    
    enum CodingKeys: String, CodingKey {
        case id, workoutSessionId = "workout_session_id", exerciseId = "exercise_id"
    }
}
